ios/StripeTerminalReactNative.swift

```diff
@@ -239,10 +242,19 @@ class StripeTerminalReactNative: RCTEventEmitter, DiscoveryDelegate, BluetoothRe
         }
 
         let locationId = params["locationId"] as? String
-
-        let connectionConfig = BluetoothConnectionConfiguration(
-            locationId: locationId ?? selectedReader.locationId ?? ""
-        )
+        let autoReconnectOnUnexpectedDisconnect = params["autoReconnectOnUnexpectedDisconnect"] as? Bool ?? false
+
+        var connectionConfig: BluetoothConnectionConfiguration
+        if autoReconnectOnUnexpectedDisconnect {
+            connectionConfig = BluetoothConnectionConfiguration(
+                        locationId: locationId ?? selectedReader.locationId ?? "",
+                        autoReconnectOnUnexpectedDisconnect: true,
+                        autoReconnectionDelegate: self
+                    )
+        } else {
+            connectionConfig = BluetoothConnectionConfiguration(
+						locationId: locationId ?? selectedReader.locationId ?? "")
+        }
```

Comment:
I think this could be simplified to a single BluetoothConnectionConfiguration if you use the `autoReconnetOnUnexpectedDisconnect` variable as the value in the init, so:

```
let connectionConfig = BluetoothConnectionConfiguration(
                        locationId: locationId ?? selectedReader.locationId ?? "",
                        autoReconnectOnUnexpectedDisconnect: autoReconnectOnUnexpectedDisconnect,
                        autoReconnectionDelegate: autoReconnectOnUnexpectedDisconnect ? self : nil
```